### DISCUSSION TITLE
Are there any good solutions for providing Sphinx as a plugin?

### DISCUSSION THREAD

**Discussion by xinetzone** [zauberzeug/nicegui#2786] [Ideas / Feature Requests]
If it's possible to integrate NiceGUI directly as a plugin into Sphinx, then leveraging Sphinx's system could provide more user-friendly and enriched functionalities for document generation.

**Comment by rodja** [comment]
Thanks for bringing this up. But could you be a bit more concrete? What would you like the NiceGUI plugin for Sphinx to provide?

**Reply by xinetzone** [reply]
By leveraging [myst](https://myst-nb.readthedocs.io/en/latest/computation/execute.html)+jupyter, I want to be able to write code in a Jupyter notebook and then use Sphinx to build an interactive documentation that supports NiceGUI's interactive buttons and other components.

or Jupyter Interactive Widgets
![](https://user-images.githubusercontent.com/591645/123929339-086f6180-d98f-11eb-8ab0-c7f9661ff41e.gif)

**Reply by xinetzone** [reply]
Something similar to [jupyterlite-sphinx](https://jupyterlite-sphinx.readthedocs.io/en/latest/).

**Reply by xinetzone** [reply]
It could also involve implementing certain Sphinx directives or roles to achieve something similar to [jupyter-execute](https://ipywidgets.readthedocs.io/en/stable/embedding.html).
```rst
.. jupyter-execute::

  from ipywidgets import VBox, jsdlink, IntSlider, Button
  s1, s2 = IntSlider(max=200, value=100), IntSlider(value=40)
  b = Button(icon='legal')
  jsdlink((s1, 'value'), (s2, 'max'))
  VBox([s1, s2, b])
  ```
